app/src/main/java/com/soen490chrysalis/papilio/view/UserProfileActivity.kt

```diff
@@ -21,12 +43,20 @@ class UserProfileActivity : AppCompatActivity()
         val view = binding.root
         setContentView(view)
 
+        val userRepository : IUserRepository =
+            UserRepository(FirebaseAuth.getInstance(), userService = UserApi.retrofitService)
+
+        userProfileViewModel = UserProfileViewModel(userRepository);
+        userProfileViewModel.getUserByFirebaseId()
+
         // Get the last signed in Google account so we can check if the user logged in with Google or through normal log in
         val acct = GoogleSignIn.getLastSignedInAccount(this)
```

Comment:
The firebase object has an attribute telling you how the user signed in (i.e. password, google, facebook, etc.). I think you should use that instead.